src/main/java/com/browserstack/automate/ci/common/proxysettings/JenkinsProxySettings.java

```diff
@@ -1,77 +1,222 @@
 package com.browserstack.automate.ci.common.proxysettings;
 
+import com.browserstack.automate.ci.common.Tools;
 import hudson.ProxyConfiguration;
+import hudson.Util;
 import jenkins.model.Jenkins;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.PrintStream;
 import java.net.InetSocketAddress;
 import java.net.Proxy;
+import java.net.URL;
+
+import static com.browserstack.automate.ci.common.logger.PluginLogger.log;
+import static com.browserstack.automate.ci.common.logger.PluginLogger.logError;
 
 public class JenkinsProxySettings {
 
     private static final ProxyConfiguration jenkinsProxy = Jenkins.getInstanceOrNull() != null ? Jenkins.getInstanceOrNull().proxy : null;
-    private static final String protocol = "https";
-    private static final String systemProxyHost = System.getProperty(protocol + ".proxyHost");
-    private static final int systemProxyPort = Integer.parseInt(System.getProperty(protocol + ".proxyPort", "0"));
-    private static final String systemProxyUser = System.getProperty(protocol + ".proxyUser");
-    private static final String systemProxyPassword = System.getProperty(protocol + ".proxyPassword");
-
-    public static Proxy getJenkinsProxy() {
-        if (hasSystemProxy()) {
-            return new Proxy(Proxy.Type.HTTP, new InetSocketAddress(systemProxyHost, systemProxyPort));
-        }
 
-        if (jenkinsProxy == null) return null;
-        final String proxyHost = jenkinsProxy.name;
-        final int proxyPort = jenkinsProxy.port;
-        return (proxyHost != null && proxyPort != 0) ? new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, proxyPort)) : null;
+    private static final String jarProxyHost = System.getProperty("https.proxyHost");
+    private static final int jarProxyPort = Integer.parseInt(System.getProperty("https.proxyPort", "443"));
```

Comment:
Default port as 443?

Reply:
Should be some value..so added this (considering `https` tags there. Though, `https` doesn't mean I can't add http proxy there). Should I change it to 3128? or any other value?

Reply:
As long as it is there to maintain that the code doesn't break. We can choose any default port. Whatever seems fit.

Reply:
Alright. Resolving then.